File: dotnet/AeronTransport.cs
namespace Epoch;

public sealed class AeronTransport : ITransport
{
    public sealed record AeronConfig(string Channel, int StreamId, string AeronDirectory);

    public AeronTransport(AeronConfig config)
    {
        Config = config;
    }

    public AeronConfig Config { get; }

    public void Send(Message message)
    {
        throw new NotSupportedException("Aeron transport not linked");
    }

    public List<Message> Poll(int max)
    {
        throw new NotSupportedException("Aeron transport not linked");
    }

    public void Close()
    {
    }
}

File: cpp/src/aeron_transport.cpp
#include "epoch/aeron_transport.h"

#include <stdexcept>

namespace epoch {

AeronTransport::AeronTransport(AeronConfig config) : config_(std::move(config)) {}

void AeronTransport::send(const Message &)
{
    throw std::runtime_error("Aeron transport not linked");
}

std::vector<Message> AeronTransport::poll(std::size_t)
{
    throw std::runtime_error("Aeron transport not linked");
}

void AeronTransport::close()
{
    closed_ = true;
}

} // namespace epoch
